lib/workers/repository/update/branch/execute-post-upgrade-commands.ts

```diff
@@ -61,6 +66,38 @@ export async function postUpgradeCommandsExecutor(
         }
       }
 
+      let dataFilePath = '';
+      if (dataFileTemplate) {
+        const dataFileContent = compile(
+          dataFileTemplate,
+          mergeChildConfig(config, upgrade),
+        );
+        logger.debug(
+          { dataFileTemplate },
+          'Processed post-upgrade commands data file template.',
+        );
+
+        const dataFileName = `post-upgrade-data-file-${crypto.randomBytes(8).toString('hex')}.tmp`;
+        dataFilePath = upath.join(privateCacheDir(), dataFileName);
+
+        try {
+          await writeSystemFile(dataFilePath, dataFileContent);
```

Comment:
It'd be safer to use outputCacheFile instead, right?:
```suggestion
          await outputCacheFile(dataFilePath, dataFileContent);
```